numba/cuda/tests/cudadrv/test_linker.py

```diff
@@ -85,18 +85,27 @@ def foo(x, y):
 
         self.assertTrue(A[0] == 123 + 2 * 321)
 
+    @require_context
+    def test_set_registers_no_max(self):
+        """Ensure that the jitted kernel used in the test_set_registers_* tests
+        uses more than 57 registers - this ensures that test_set_registers_*
+        are really checking that they reduced the number of registers used from
+        something greater than the maximum."""
+        compiled = cuda.jit(function_with_lots_of_registers)
+        compiled = compiled.specialize(np.empty(32), *range(6))
+        self.assertGreater(compiled._func.get().attrs.regs, 57)
+
     @require_context
     def test_set_registers_57(self):
         compiled = cuda.jit(max_registers=57)(function_with_lots_of_registers)
         compiled = compiled.specialize(np.empty(32), *range(6))
-        self.assertEquals(57, compiled._func.get().attrs.regs)
+        self.assertLessEqual(compiled._func.get().attrs.regs, 57)
 
     @require_context
     def test_set_registers_38(self):
         compiled = cuda.jit(max_registers=38)(function_with_lots_of_registers)
         compiled = compiled.specialize(np.empty(32), *range(6))
-        self.assertEquals(38, compiled._func.get().attrs.regs)
-
+        self.assertLessEqual(compiled._func.get().attrs.regs, 38)
```

Comment:
think this was perhaps covered a bit by another PR, so there might be a minor conflict.

Reply:
a merge made it disappear.